### PR TITLE
Add support for PageLabels in the viewer (issue 6902, bug 793632)

### PR DESCRIPTION
**Pull request by Snuffleupagus**
_Please refer to the individual commit messages._

Can be tested using e.g. the following files: http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/PDF32000_2008.pdf, or http://mirrors.ctan.org/info/lshort/english/lshort.pdf.
Fixes issue #6902 and [bug 793632](https://bugzilla.mozilla.org/show_bug.cgi?id=793632).

**TODO:**
- [x] Localize the new strings, and cleanup the now obsolete ones.

### CODE REVIEW

**Comment by timvandermeij** [comment]
Fixes #3683 too.

**Comment by timvandermeij** [comment]
Because we do not have the "Page" label anymore, we can adjust the CSS a bit to make sure nothing overlaps anymore during resize:

Change
<pre>
  html[dir='ltr'] .outerCenter {
    float: left;
    left: 205px;
  }
  html[dir='rtl'] .outerCenter {
    float: right;
    right: 205px;
  }
</pre>

to

<pre>
  html[dir='ltr'] .outerCenter {
    float: left;
    left: 160px;
  }
  html[dir='rtl'] .outerCenter {
    float: right;
    right: 160px;
  }
</pre>

(i.e., 205 to 160 pixels) so that when you resize the window, the zoom controls don't move anymore and the icons do not overlap the select box anymore.

**Comment by timvandermeij** [comment]
I think overall this is a great solution and most in line with how other viewers (especially Adobe Reader) handle it.

@yurydelendik @brendandahl What do you think about how the page labels are handled in the UI with this patch? If you also think that this is the way to go, I will review this patch in more detail.

**Comment by yurydelendik** [comment]
Can we block this by extracting a toolbar control from app.js?

File: web/app.js [hunk 1]
```diff
@@ -323,6 +325,9 @@ var PDFViewerApplication = {
     });
     this.pdfSidebar.onToggled = this.forceRendering.bind(this);
 
+    this.pageNumberInput = document.getElementById('pageNumber');
```
Comment (timvandermeij) [inline web/app.js]: The `getElementById` calls in this PR probably need to be moved to the viewer configuration part after #7209.

**Comment by timvandermeij** [comment]
@yurydelendik Sounds good to me. Perhaps the patch will be slightly easier too when the toolbar code is more centralized.

File: web/app.js (hunk 1 above)
Comment (Snuffleupagus) [inline web/app.js]: Sure, but this patch has already been bit-rotten a couple of times, and until I get an OK on the feature itself *and* the general implementation, it doesn't make much sense for me to put in the time required to rebase this yet again.
Getting past the `viewer.js` -> `app.js` rename was "fun" enough :-)

**Comment by Snuffleupagus** [comment]
> Can we block this by extracting a toolbar control from app.js?

I've now rebased/rewritten this PR on top of PR #7458 (which at least improved the current situation wrt updating the toolbar); is this enough that we can now consider moving forward with this PR?

**Comment by timvandermeij** [comment]
I would say so, yes. I see no more reason to block this work. This patch actually turned out to have less code than I initially thought we would need, so that is nice too!

**Comment by timvandermeij** [comment]
@Snuffleupagus Do you want to address https://github.com/mozilla/pdf.js/pull/6945#issuecomment-177595111 in a separate commit too? It would be nice to fix the responsiveness issues as well, which this patch already partially fixes.

**Comment by Snuffleupagus** [comment]
> I would say so, yes. I see no more reason to block this work. This patch actually turned out to have less code than I initially thought we would need, so that is nice too!

Thanks; I've actually tried to reduced the size of the total diff as best as I could.
I suppose that one of the remaining things we need to do, is to get consensus that it's actually OK to remove the "Page:" label from the toolbar.

> @Snuffleupagus Do you want to address #6945 (comment) in a separate commit too? It would be nice to fix the responsiveness issues as well, which this patch already partially fixes.

I think I'd prefer if that was done in a follow-up instead, since it might not be as easy as it first seems.

Keep in mind that even without the "Page:" label in the toolbar, for documents that use Page Labels, some (or even all) of the space saving could actually be offset by the longer `numPages` label.
This is obviously dependent on the number of pages in the document (e.g. consider cases where numPages > 1000), and also what locale is being using (since length of the "of" label differs between languages).
*Note:* Even in English, the space saving isn't necessarily that great, this is how it looks with [lshort.pdf](http://mirrors.ctan.org/info/lshort/english/lshort.pdf):

page-label-diff

**Comment by Snuffleupagus** [comment]
Can we try and land this feature, obviously after it has properly reviewed,  while PDF.js is still shipping in Firefox? ;-)
(Considering that PDFium, at least as it appears in Chrome, doesn't seem to support this feature it'd be nice if we added this to PDF.js IMHO.)

**Comment by yurydelendik** [comment]
> Can we try and land this feature?

Yes, let's try to do it this cycle.

**Comment by yurydelendik** [comment]
>  Per IRC: @timvandermeij, I only need to know if we have agreement on removing the `Page:` label and replacing it with a tooltip. Personally I think this change is fine because it makes things more in line with other viewers and the extra space is required for the page labels.

Sounds good: we can remove it.

**Comment by timvandermeij** [comment]
Amazing work, and the implementation is really clean!

**Pull request merged**